### PR TITLE
Pensions other expenses require fields

### PR DESCRIPTION
**Pull request by cehsu**
screen shot 2017-08-01 at 4 08 45 pm

screen shot 2017-08-01 at 4 52 20 pm

screen shot 2017-08-01 at 4 53 04 pm

### CODE REVIEW

**Comment by jbalboni** [comment]
@cehsu I'm planning on waiting until you get the schema changes merged in here to review this. If that doesn't make sense, let me know.

**Comment by cvalarida** [comment]
Is there an issue this is connected to? If not, what's the background? (I'm assuming this came from the stakeholder review we had the other day, but I'm not 100% on that.)

**Comment by cehsu** [comment]
Thanks @jbalboni @cvalarida, the schema changes have been merged, right now there's a bug that disables submit (select yes, fill out fields, cannot submit). Is the cause for this clear to either of you?

**Comment by jbalboni** [comment]
@cehsu Looking now.

**Comment by jbalboni** [comment]
@cehsu The schema is messed up, looks like (we should have caught that on review):

This has the required fields in the wrong place:

```
"otherExpenses": {
      "type": "array",
      "minItems": 1,
      "required": [
        "amount",
        "purpose",
        "paidTo",
        "date"
      ],
      "items": {
        "type": "object",
        "properties": {
          "amount": {
            "type": "integer"
          },
          "purpose": {
            "type": "string"
          },
          "paidTo": {
            "type": "string"
          },
          "date": {
            "$ref": "#/definitions/date"
          }
        }
      }
    },
```

It should be:

```
"otherExpenses": {
      "type": "array",
      "minItems": 1,
      
      "items": {
        "type": "object",
        "required": [
          "amount",
          "purpose",
          "paidTo",
          "date"
        ],
        "properties": {
          "amount": {
            "type": "integer"
          },
          "purpose": {
            "type": "string"
          },
          "paidTo": {
            "type": "string"
          },
          "date": {
            "$ref": "#/definitions/date"
          }
        }
      }
    },
```

**Comment by cehsu** [comment]
Thanks @jbalboni will fix now.

File: src/js/pensions/config/form.js
Comment (jbalboni) [inline src/js/pensions/config/form.js]: You shouldn't need these changes now.

File: src/js/pensions/config/form.js
Comment (jbalboni) [inline src/js/pensions/config/form.js]: Same as above

File: src/js/pensions/config/form.js
Comment (jbalboni) [inline src/js/pensions/config/form.js]: Here, too.

**Review by cvalarida — APPROVED** [review]

**Pull request merged**
